addons/website/static/src/snippets/s_dynamic_filter_carousel/000.js

```diff
@@ -0,0 +1,122 @@
+odoo.define('website.s_dynamic_filter_carousel', function (require) {
+'use strict';
+
+const config = require('web.config');
+const core = require('web.core');
+const publicWidget = require('web.public.widget');
+
+const qweb = core.qweb;
+
+const DynamicFilterCarousel = publicWidget.Widget.extend({
+    selector: '.s_dynamic_filter_carousel',
+    xmlDependencies: ['/website/static/src/xml/website.s_dynamic_filter_carousel.xml'],
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     */
+    init: function () {
+        this._super.apply(this, arguments);
+        this.uniqueId = _.uniqueId('s_dynamic_filter_carousel_');
+        this._onResizeChange = _.debounce(this._refresh_carousel_content, 100);
```

Comment:
can be smarter and keep last size. 
It will avoid new rpc, new render and ... if we keep same ratio / deviceSize

And btw 100 is very too small imo, open your network tab and resize yourself your window, you will have a lot of request, a least 5 * (1+number of item) :/